src/AuthorizationUtils.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Bundle\MercureBundle;
+
+use Symfony\Component\HttpFoundation\Cookie;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+use Symfony\Component\Mercure\Exception\RuntimeException;
+use Symfony\Contracts\Service\ServiceProviderInterface;
+
+final class AuthorizationUtils
+{
+    private const MERCURE_AUTHORIZATION_COOKIE_NAME = 'mercureAuthorization';
+
+    private $defaultHub;
+    private $hubs;
+    private $factories;
+
+    public function __construct(string $defaultHub, ServiceProviderInterface $hubs, ServiceProviderInterface $factories)
+    {
+        $this->defaultHub = $defaultHub;
+        $this->hubs = $hubs;
+        $this->factories = $factories;
+    }
+
+    /**
+     * Create Authorization cookie for the given hub.
+     *
+     * @param string[]    $subscribe        a list of topics that the authorization cookie will allow subscribing to
+     * @param string[]    $publish          a list of topics that the authorization cookie will allow publishing to
+     * @param string|null $hub              the hub to generate the cookie for
+     * @param mixed[]     $additionalClaims an array of additional claims for the JWT
+     */
+    public function createCookie(Request $request, array $subscribe = [], array $publish = [], ?string $hub = null, array $additionalClaims = []): Cookie
```

Comment:
I would put the `$hub` parameter at the end of the list. It's probably the one that will be used least frequently, also I think it's more coherent to group together all JWT-related parameters.